File: analyze_excel.py

import pandas as pd
import json

excel_file = 'Excel Loja.xlsm'

try:
    xl = pd.ExcelFile(excel_file)
    sheets = xl.sheet_names
    print(f"Sheet names: {sheets}")
    
    summary = {}
    for sheet in sheets[:5]: # Analyze first 5 sheets
        df = pd.read_excel(excel_file, sheet_name=sheet, nrows=5)
        summary[sheet] = {
            "columns": df.columns.tolist(),
            "preview": df.to_dict(orient='records')
        }
    
    with open('excel_analysis.json', 'w', encoding='utf-8') as f:
        json.dump(summary, f, indent=2, ensure_ascii=False)
    print("Analysis saved to excel_analysis.json")

except Exception as e:
    print(f"Error: {e}")

File: analyze_excel.js

const XLSX = require('xlsx');
const fs = require('fs');

const excelFile = 'Excel Loja.xlsm';

try {
    const workbook = XLSX.readFile(excelFile);
    const sheetNames = workbook.SheetNames;
    console.log(`Sheet names: ${sheetNames}`);

    const summary = {};
    sheetNames.slice(0, 10).forEach(sheetName => {
        const worksheet = workbook.Sheets[sheetName];
        const data = XLSX.utils.sheet_to_json(worksheet, { header: 1, range: 0 }); // Read top rows
        
        if (data.length > 0) {
            summary[sheetName] = {
                columns: data[0],
                preview: data.slice(1, 4) // First 3 rows of data
            };
        }
    });

    fs.writeFileSync('excel_analysis.json', JSON.stringify(summary, null, 2));
    console.log("Analysis saved to excel_analysis.json");

} catch (e) {
    console.error(`Error: ${e.message}`);
}
